frontend/src/components/common/StripeCustomer.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState } from "react";
+import dotenv from "dotenv";
+import { loadStripe } from "@stripe/stripe-js";
+import {
+  Elements,
+  CardElement,
+  useStripe,
+  useElements,
+} from "@stripe/react-stripe-js";
+
+dotenv.config();
+
+const stripePromise = loadStripe(process.env.REACT_APP_STRIPE_PUBLISHABLE_KEY as string);
+
+const CreateCustomerForm: React.FC = () => {
+  const [name, setName] = useState("");
+  const [email, setEmail] = useState("");
+  const stripe = useStripe();
+  const elements = useElements();
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!stripe || !elements) {
+      return;
+    }
+
+    const cardElement = elements.getElement(CardElement);
+
+    if (!cardElement) {
+      console.error("CardElement not found");
+      return;
+    }
+
+    const {
+      error: paymentMethodError,
+      paymentMethod,
+    } = await stripe.createPaymentMethod({
+      type: "card",
+      card: cardElement,
+      billing_details: {
+        name,
+      },
+    });
+
+    console.log("Payment Method object:", paymentMethod);
+
+    if (paymentMethodError) {
+      console.error("Error creating payment method:", paymentMethodError);
+    } else {
+      try {
+        const response = await fetch(
```

Comment:
did you test that this frontend component can successfully create a Stripe customer?

when I tried directly hitting the `/create-customer` endpoint in the backend, it looks like I require a Stripe bearer token
```de_ts_backend  | {"level":"error","message":"[/app/services/implementations/stripeService.ts] Error creating a 
customer for user Test","timestamp":"2024-07-24T18:23:41.581Z"}
de_ts_backend  | StripeAuthenticationError: You did not provide an API key. You need to provide your API key in the Authorization header, using Bearer auth (e.g. 'Authorization: Bearer YOUR_SECRET_KEY'). See https://stripe.com/docs/api#authentication for details, or we can help at https://support.stripe.com/.
de_ts_backend  |     at res.toJSON.then.Error_js_1.StripeAPIError.message (/app/node_modules/stripe/cjs/RequestSender.js:96:31)
de_ts_backend  |     at processTicksAndRejections (node:internal/process/task_queues:95:5) {
de_ts_backend  |   type: 'StripeAuthenticationError',
de_ts_backend  |   raw: {
de_ts_backend  |     message: "You did not provide an API key. You need to provide your API key in the Authorization header, using Bearer auth (e.g. 'Authorization: Bearer YOUR_SECRET_KEY'). See https://stripe.com/docs/api#authentication for details, or we can help at https://support.stripe.com/.",
de_ts_backend  |     type: 'invalid_request_error',
de_ts_backend  |     headers: {
de_ts_backend  |       server: 'nginx',
de_ts_backend  |       date: 'Wed, 24 Jul 2024 18:23:41 GMT',
de_ts_backend  |       'content-type': 'application/json',
de_ts_backend  |       'content-length': '342',
de_ts_backend  |       connection: 'keep-alive',
de_ts_backend  |       'access-control-allow-credentials': 'true',
de_ts_backend  |       'access-control-allow-methods': 'GET,HEAD,PUT,PATCH,POST,DELETE',
de_ts_backend  |       'access-control-allow-origin': '*',
de_ts_backend  |       'access-control-expose-headers': 'Request-Id, Stripe-Manage-Version, Stripe-Should-Retry, X-Stripe-External-Auth-Required, X-Stripe-Privileged-Session-Required',
de_ts_backend  |       'access-control-max-age': '300',
de_ts_backend  |       'cache-control': 'no-cache, no-store',
de_ts_backend  |       vary: 'Origin',
de_ts_backend  |       'www-authenticate': 'Basic realm="Stripe"',
de_ts_backend  |       'strict-transport-security': 'max-age=63072000; includeSubDomains; preload'
de_ts_backend  |     },
de_ts_backend  |     statusCode: 401,
de_ts_backend  |     requestId: undefined
de_ts_backend  |   },
de_ts_backend  |   rawType: 'invalid_request_error',
de_ts_backend  |   code: undefined,
de_ts_backend  |   doc_url: undefined,
de_ts_backend  |   param: undefined,
de_ts_backend  |   detail: undefined,
de_ts_backend  |   headers: {
de_ts_backend  |     server: 'nginx',
de_ts_backend  |     date: 'Wed, 24 Jul 2024 18:23:41 GMT',
de_ts_backend  |     'content-type': 'application/json',
de_ts_backend  |     'content-length': '342',
de_ts_backend  |     connection: 'keep-alive',
de_ts_backend  |     'access-control-allow-credentials': 'true',
de_ts_backend  |     'access-control-allow-methods': 'GET,HEAD,PUT,PATCH,POST,DELETE',
de_ts_backend  |     'access-control-allow-origin': '*',
de_ts_backend  |     'access-control-expose-headers': 'Request-Id, Stripe-Manage-Version, Stripe-Should-Retry, 
X-Stripe-External-Auth-Required, X-Stripe-Privileged-Session-Required',
de_ts_backend  |     'access-control-max-age': '300',
de_ts_backend  |     'cache-control': 'no-cache, no-store',
de_ts_backend  |     vary: 'Origin',
de_ts_backend  |     'www-authenticate': 'Basic realm="Stripe"',
de_ts_backend  |     'strict-transport-security': 'max-age=63072000; includeSubDomains; preload'
de_ts_backend  |   },
de_ts_backend  |   requestId: undefined,
de_ts_backend  |   statusCode: 401,
de_ts_backend  |   charge: undefined,
de_ts_backend  |   decline_code: undefined,
de_ts_backend  |   payment_intent: undefined,
de_ts_backend  |   payment_method: undefined,
de_ts_backend  |   payment_method_type: undefined,
de_ts_backend  |   setup_intent: undefined,
de_ts_backend  |   source: undefined
de_ts_backend  | }
```

but I don't see a bearer token being passed when we call the `/create-customer` endpoint from our frontend here

Reply:
also, would you be able to include a sample successful response from the `/create-customer` endpoint if possible 🙏 

just curious what it would look like

Reply:
Matt and I reviewed in person - sample successful response screenshot attached in description (console output)